tests/e2e/test_rosacli_cluster.go

```diff
@@ -1463,3 +1462,284 @@ var _ = Describe("Create sts and hcp cluster with the IAM roles with path settin
+				rosaClient.Runner.UnsetArgs()
+				_, err := ocmResourceService.DeleteAccountRole("--mode", "auto",
+					"--prefix", accountRolePrefix,
+					"-y")
+				Expect(err).To(BeNil())
+			}
+
+		})
+
+		It("to validate to create cluster with existing operator roles prefix - [id:45742]",
+			labels.Critical, labels.Runtime.Day1Supplemental,
+			func() {
+				By("Create acount-roles")
+				accountRolePrefix = "testAr45742"
+				output, err := ocmResourceService.CreateAccountRole(
+					"--mode", "auto",
+					"--prefix", accountRolePrefix,
+					"-y")
+				Expect(err).To(BeNil())
+				textData := rosaClient.Parser.TextData.Input(output).Parse().Tip()
+				Expect(textData).To(ContainSubstring("Created role"))
+
+				arl, _, err := ocmResourceService.ListAccountRole()
+				Expect(err).To(BeNil())
+				ar := arl.DigAccountRoles(accountRolePrefix, false)
+
+				By("Create one sts cluster")
+				clusterNameToClean = "test-45742"
+				operatorRolePreifx := "opPrefix45742"
+				_, err, _ = clusterService.Create(
+					clusterNameToClean, "--sts",
+					"--mode", "auto",
+					"--role-arn", ar.InstallerRole,
+					"--support-role-arn", ar.SupportRole,
+					"--controlplane-iam-role", ar.ControlPlaneRole,
+					"--worker-iam-role", ar.WorkerRole,
+					"--operator-roles-prefix", operatorRolePreifx,
+					"-y",
+				)
+				Expect(err).To(BeNil())
+
+				By("Create another cluster with the same operator-roless-prefix")
+				clusterName := "test-45742b"
+				out, err, _ := clusterService.Create(
+					clusterName, "--sts",
+					"--mode", "auto",
+					"--role-arn", ar.InstallerRole,
+					"--support-role-arn", ar.SupportRole,
+					"--controlplane-iam-role", ar.ControlPlaneRole,
+					"--worker-iam-role", ar.WorkerRole,
+					"--operator-roles-prefix", operatorRolePreifx,
+					"-y",
+				)
+				Expect(err).NotTo(BeNil())
+				Expect(out.String()).To(ContainSubstring("already exists"))
+				Expect(out.String()).To(ContainSubstring("provide a different prefix"))
+			})
+	})
+
+var _ = Describe("create/delete operator-roles and oidc-provider to cluster",
+	labels.Feature.Cluster, func() {
+		defer GinkgoRecover()
+		var (
+			rosaClient *rosacli.Client
+
+			accountRolePrefix  string
+			ocmResourceService rosacli.OCMResourceService
+			clusterNameToClean string
+			clusterService     rosacli.ClusterService
+			clusterID          string
+		)
+
+		BeforeEach(func() {
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			ocmResourceService = rosaClient.OCMResource
+			clusterService = rosaClient.Cluster
+		})
+
+		AfterEach(func() {
+
+			By("Delete cluster")
+			rosaClient.Runner.UnsetArgs()
+			clusterListout, err := clusterService.List()
+			Expect(err).To(BeNil())
+			clusterList, err := clusterService.ReflectClusterList(clusterListout)
+			Expect(err).To(BeNil())
+
+			if clusterList.IsExist(clusterID) {
+				rosaClient.Runner.UnsetArgs()
+				_, err = clusterService.DeleteCluster(clusterID, "-y")
+				Expect(err).To(BeNil())
+			}
```

Comment:
Can not we delete the cluster directly, and pass the deletion if it is 200 and 404?

Reply:
@yasun1 In the testing step, deleting cluster is as a step, the deleting cluster step in AfterEach is for the failure at the deletion step during scenario is failed.